bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpClientFactory.java

```diff
@@ -0,0 +1,50 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.io.net.http;
+
+import org.eclipse.jetty.client.HttpClient;
+
+/**
+ * Factory class to create jetty http clients
+ *
+ * @author Michael Bock - initial API
+ */
+public interface HttpClientFactory {
+
+    /**
+     * Creates a new jetty http client..
+     * The returned client is already started but must be stopped
+     * by the consumer after usage. The client lifecycle should be the same as for the consumer.
+     * DO NOT CREATE NEW CLIENTS FOR EACH REQUEST!
+     * 
+     * @param consumerName the for identifying the consumer in the jetty thread pool.
+     *            Must be between 4 and 20 characters long and must contain only the following characters [a-zA-Z0-9-_]
+     * @param endpoint the desired endpoint, protocol and host are sufficient
+     * @return the jetty client
+     * @throws NullPointerException, IllegalArgumentException
+     */
+    HttpClient getHttpClient(String consumerName, String endpoint);
```

Comment:
Just a proposal: I believe it is more conventional to name the methods - `newHttpClient(..)` but a philosophical proposal. 😉